vignettes/cpp11_faq.Rmd

```diff
@@ -0,0 +1,184 @@
+---
+title: "cpp11 FAQ"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{cpp11 FAQ}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Intended to answer common questions about the usage of cpp11 objects
+
+## FAQ
+
+* How do I modify a vector in place?
+
+  * Using `writable::` will always make a copy, but it has a move constructor, so you can use `cpp11::writable::integers(std::move(x))` and it won't make a copy of the data.
```

Comment:
I think we should move this one down, I generally think doing this is a mistake many of the times, so don't want to make it too prominent.

Reply:
Sounds good!